cmd/leveldbbench/leveldbbench.go

```diff
@@ -0,0 +1,622 @@
+			ro.Strict = opt.StrictAll
+		} else {
+			ro.Strict = opt.DefaultStrict
+		}
+		if *nilReadOptions {
+			ro = nil
+		}
+		var start time.Time
+		trs := make([]*TestResult, 0)
+
+		sequentialWritesDesc := "random"
+		if *sequentialWrites {
+			sequentialWritesDesc = "sequential"
+		}
+		sequentialReadsDesc := "random"
+		if *sequentialReads {
+			sequentialReadsDesc = "sequential"
+		}
+
+		if *fullScan {
+			start = time.Now()
+			opCount, valueDist := runFullScan(ctx, db, &wo, ro)
+			tr := NewTestResult(start, time.Now(), "full scan", opCount, db)
+			tr.ValueDist = valueDist
+			trs = append(trs, tr)
+			return printSummary(trs)
+		}
+
+		// in no write mode, we assume the database as already been populated in a previous run or we're using some other database
+		if !*readOnly {
+			start = time.Now()
+			writeData(ctx, db, &wo, 0, *writeLimit, *sequentialWrites)
+			trs = append(trs, NewTestResult(start, time.Now(), fmt.Sprintf("initial %s write", sequentialWritesDesc), *writeLimit, db))
+
+			for i := 0; i < int(*overwriteCount); i += 1 {
+				start = time.Now()
+				writeData(ctx, db, &wo, 0, *writeLimit, *sequentialWrites)
+				trs = append(trs, NewTestResult(start, time.Now(), fmt.Sprintf("%s overwrite %d", sequentialWritesDesc, i), *writeLimit, db))
+			}
+
+			start = time.Now()
+			runFullCompact(ctx, db, &wo)
+			trs = append(trs, NewTestResult(start, time.Now(), "compaction", 1, db))
+		}
+
+		if *sequentialReads {
+			start = time.Now()
+			readSeq(ctx, db, &wo, *readLimit)
+			trs = append(trs, NewTestResult(start, time.Now(), fmt.Sprintf("%s read", sequentialReadsDesc), *readLimit, db))
+		} else {
+			start = time.Now()
+			readRandom(ctx, db, ro, *readLimit)
+			trs = append(trs, NewTestResult(start, time.Now(), fmt.Sprintf("%s read", sequentialWritesDesc), *readLimit, db))
+		}
+
+		log.Info().Msg("Close DB")
+		err = db.Close()
+		if err != nil {
+			log.Error().Err(err).Msg("error while closing db")
+		}
+
+		return printSummary(trs)
+	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		var err error
+		sizeDistribution, err = parseRawSizeDistribution(*rawSizeDistribution)
+		if err != nil {
+			return err
+		}
+		if *keySize > 64 {
+			return fmt.Errorf(" max supported key size is 64 bytes. %d is too big", *keySize)
+		}
+		return nil
+	},
+}
+
+func printSummary(trs []*TestResult) error {
+	jsonResults, err := json.Marshal(trs)
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(jsonResults))
+	return nil
+}
+
+func runFullCompact(ctx context.Context, db *leveldb.DB, wo *opt.WriteOptions) {
+	err := db.CompactRange(util.Range{Start: nil, Limit: nil})
+	if err != nil {
+		log.Fatal().Err(err).Msg("error compacting data")
+	}
+}
+func runFullScan(ctx context.Context, db *leveldb.DB, wo *opt.WriteOptions, ro *opt.ReadOptions) (uint64, []uint64) {
+	pool := make(chan bool, *degreeOfParallelism)
+	var wg sync.WaitGroup
+	// 32 should be safe here. That would correspond to a single value that's 4.2 GB
```

Comment:
> That would correspond to a single value that's 4.2 GB

curious on what this comment means 👀

Reply:
We are basically taking the length of the value then getting the bit length of that length. So a bit length of 32 supports an underlying value of 4gb which is larger than I suspect we need. But we could always increase.